e2e/devtools.test.js

```diff
@@ -0,0 +1,385 @@
+    await browser.navigateTo('http://guinea-pig.webdriver.io')
+    expect(await browser.getTitle()).toBe('WebdriverJS Testpage')
+    expect(await browser.getUrl()).toContain('http://guinea-pig.webdriver.io')
+    expect(await browser.getPageSource()).toContain('WebdriverJS Testpage')
+})
+
+describe('timeouts', () => {
+    beforeAll(async () => {
+        await browser.navigateTo('http://guinea-pig.webdriver.io')
+    })
+
+    it('getTimeouts', async () => {
+        expect(await browser.getTimeouts()).toEqual({
+            implicit: 0,
+            pageLoad: 300000,
+            script: 30000
+        })
+    })
+
+    it('setTimeouts', async () => {
+        await browser.setTimeouts(1, 2, 3)
+        expect(await browser.getTimeouts()).toEqual({
+            implicit: 1,
+            pageLoad: 2,
+            script: 3
+        })
+    })
+
+    it('test script timeout', async () => {
+        expect.assertions(2)
+
+        const asyncCommand = () => browser.executeAsyncScript(
+            'return setTimeout(() => arguments[2](document.title + \' \' + arguments[0] + arguments[1]), 100)',
+            ['Test', '!'])
+
+        try {
+            await asyncCommand()
+        } catch (e) {
+            expect(e.message).toBe('Evaluation failed: script timeout')
+        }
+
+        await browser.setTimeouts(0, 1000, 300)
+        expect(await asyncCommand()).toBe('WebdriverJS Testpage Test!')
+    })
+
+    afterAll(async () => {
+        await browser.setTimeouts(0, 300000, 300000)
+    })
+})
+
+describe('elements', () => {
+    beforeAll(async () => {
+        await browser.navigateTo('http://guinea-pig.webdriver.io')
+    })
+
+    it('findElement with css selector', async () => {
+        const yellowBox = await browser.findElement('css selector', '.yellow')
+        expect(yellowBox).toEqual({ [ELEMENT_KEY]: 'ELEMENT-1' })
+        const redBox = await browser.findElement('css selector', '.red')
+        expect(redBox).toEqual({ [ELEMENT_KEY]: 'ELEMENT-2' })
+    })
+
+    it('find nested element with css selector', async () => {
+        const nested = await browser.findElement('css selector', '.nested')
+        const header = await browser.findElementFromElement(nested[ELEMENT_KEY], 'css selector', '.findme')
+        expect(header).toEqual({ [ELEMENT_KEY]: 'ELEMENT-4' })
+    })
+
+    it('findElement with xPath', async () => {
+        const xPathElement = await browser.findElement('xpath', '//*[@id="newWindow"]')
+        expect(xPathElement).toEqual({ [ELEMENT_KEY]: 'ELEMENT-5' })
+    })
+
+    it('findElements with css selector', async () => {
+        const boxes = await browser.findElements('css selector', '.box')
+        expect(boxes).toHaveLength(5)
+    })
+
+    it('find nested elements with css selector', async () => {
+        const nested = await browser.findElement('css selector', '.nested')
+        const spans = await browser.findElementsFromElement(nested[ELEMENT_KEY], 'css selector', 'span')
```

Comment:
what do you think on adding `findElementsFromElement` for xpath strategy as well?